src/XRP/issued-currency-client.ts

```diff
@@ -793,13 +793,12 @@ export default class IssuedCurrencyClient {
     amount: string,
   ): Promise<TransactionResult> {
     const issuer = sender.getAddress()
-    return await this.issuedCurrencyPayment(
-      sender,
-      destination,
+    const issuedCurrency = {
       currency,
       issuer,
-      amount,
-    )
+      value: amount,
+    }
+    return await this.issuedCurrencyPayment(sender, destination, issuedCurrency)
```

Comment:
I'm sure this is passing the typescript checks, but I'm confused by something. If this is supposed to return a `Promise<TransactionResult>` then wouldn't you `return this.issuesCurrencyPayment...`, IE without `await`? Also, IMO it might make more sense to in fact return the promise, so that the caller can have more flexibility on when it's triggered